src/main/webapp/resources/js/pages/admin/components/announcements/CreateAnnouncementModal.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useRef } from "react";
+import { Checkbox, Form, Input, Modal } from "antd";
+import { IconEdit } from "../../../../components/icons/Icons";
+import { FONT_COLOR_PRIMARY } from "../../../../styles/fonts";
+import { MarkdownEditor } from "../../../../components/markdown/MarkdownEditor";
+
+/**
+ * React component to display an Ant Design Modal containing a form to create a new announcement
+ * @param {boolean} visible - whether the modal is open or not
+ * @param {function} closeModal - the function to close the modal
+ * @param {function} createAnnouncement - the function to create a new announcement and refresh the table
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export function CreateAnnouncementModal({
+  visible,
+  closeModal,
+  createAnnouncement,
+}) {
+  const markdownRef = useRef();
+  const [form] = Form.useForm();
+
+  const onCancel = () => {
+    form.resetFields();
+    closeModal();
+  };
+
+  function saveMarkdown() {
+    form.validateFields().then((values) => {
+      const markdown = markdownRef.current.getMarkdown();
+      const title = values.title;
+      const priority = values.priority;
+
+      createAnnouncement(title, markdown, priority);
+      closeModal();
+    });
+  }
+  return (
+    <Modal
+      visible={visible}
+      title={
+        <>
+          <IconEdit style={{ color: FONT_COLOR_PRIMARY }} />
+          {i18n("CreateNewAnnouncement.title")}
+        </>
+      }
+      width="80%"
+      okText={i18n("CreateNewAnnouncement.okBtn")}
+      okButtonProps={{
+        className: "t-submit-announcement",
+      }}
+      onOk={saveMarkdown}
+      onCancel={onCancel}
```

Comment:
You're form is not resetting when the modal is closed.  This might be a little tricky as you will need to reset the contents of the markdown as well.

Reply:
Can you also make the wording a little more descriptive.  I suck at word smithing, but even "Set as priority announcement"

Reply:
Not sure I understood this one fully, but I tried. Please see 984d376a2e036f59ae6f5da31dc6caa72853b821.

Reply:
Just change the checkbox label from "Priority" to "Set as a priority message"  or something similar.

Reply:
Oh ok. Please see 458b3dc5c8e84271f36f8cf25345b23fcb75674c.

Reply:
Hopefully, this is closer. Please see d48a28b22916603a02a375fd03370557118dd28d.

Reply:
That looks great